File: Android/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/core/SettingItem.java
package com.didichuxing.doraemonkit.kit.core;


import androidx.annotation.DrawableRes;
import androidx.annotation.StringRes;

/**
 * Created by wanglikun on 2018/9/14.
 */

public class SettingItem {
    public @StringRes
    final int desc;

    public String rightDesc;

    public @DrawableRes
    int icon;

    public boolean isChecked;

    public boolean canCheck;

    public SettingItem(@StringRes int desc) {
        this.desc = desc;
    }

    public SettingItem(@StringRes int desc, boolean isChecked) {
        this.desc = desc;
        this.isChecked = isChecked;
        this.canCheck = true;
    }

    public SettingItem(@StringRes int desc, @DrawableRes int icon) {
        this.desc = desc;
        this.icon = icon;
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/core/SettingItem.kt
package com.didichuxing.doraemonkit.kit.core

import androidx.annotation.DrawableRes
import androidx.annotation.StringRes

/**
 * Created by wanglikun on 2018/9/14.
 */
data class SettingItem(@StringRes val desc: Int,
                       @DrawableRes val icon: Int,
                       val canCheck: Boolean = true,
                       var isChecked: Boolean = false,
                       var rightDesc: String? = null){

    constructor(desc: Int):this(desc
    ,0,false,false)

    constructor(desc: Int,isChecked: Boolean):this(desc
            ,0,true,isChecked)
}